src/main/java/com/sphy/hotelmanagementapplication/repository/OrderRepository.java

```diff
@@ -7,16 +7,23 @@
 import org.springframework.stereotype.Repository;
 
 import java.time.LocalDate;
+import java.util.List;
 
 /***
  * created by gp
  */
 @Repository
 public interface OrderRepository extends CrudRepository<Order,Long> {
 
-    @Query("SELECT count(*) from orders o where o.room=?3 and" +
+    @Query("SELECT count(o) from orders o where o.room=?3 and" +
```

Comment:
Please don't use ?  for passing parameters, bacuase there might be a way that this would be vulnerable to sql injection attacks. Use the annotation @Param with named parameters, such as in method findAllAdmin

For reference: https://www.baeldung.com/jpa-query-parameters